### DISCUSSION TITLE
Is it possible to check if user owns the data from another collection ?

### DISCUSSION THREAD

**Discussion by Penkie** [pocketbase/pocketbase#841] [Q&A] [answered]
Hi everyone,

I'm currently using firebase and I'm disapointed more and more as I go forward so I was wondering some alternatives. And so I found pocketbase whiche I find it's a good alternative.

Anyway, I have two collections : `lists` and `tasks`. It's for a todo app I'm building for myself basicaly.
Lists stands for like a "category" for my tasks. A task has to be in a list so when I create a task it has a `listId`. A list has to be owned by a user.

My question is, is it possible on the creation of a task, to verify that the listId given is owned by the user that do the action ?
If so, what's the rule syntaxe ?

Thanks in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
You can create a `user` type field in the `lists` collection, eg. `owner` and then for create and update API rules in your tasks collection you can use something like:

```
@request.user.id != "" && @collection.lists.owner = @request.user.id && @collection.lists.id = @request.data.listId
```

---

Notes:

- This weekend there will be a v0.8 pre-release where `users` and `profiles` will be merged per #376.

- Technically the create rule could be simplified to just:
   ```
   @request.user.id != "" && listId.owner = @request.user.id
   ```
   But the above will not work for update because in that case the document fields store the old value, where `@request.data.*` is always the submitted ones.
   In the future we'll have support for querying relation fields through the `@request.data.*` filter so that the create and update rule would be simplified to just:
   ```
   @request.user.id != "" && @request.data.listId.owner = @request.user.id
   ```

**Reply by Penkie** [reply]
This first example would work both for list, create and update ?

**Reply by ganigeorgiev** [reply]
@Penkie Yes, the first example rule should work for both **create** and **update** of a `tasks` record.

**Reply by ganigeorgiev** [reply]
@Penkie For List, View and Delete `tasks` API Rules, you could use the following rule:
```
@request.user.id != "" && listId.owner = @request.user.id
```

**Reply by Penkie** [reply]
Thanks for your replies, it helps a lot !
I have another quick question if you don't mind. 
I'm doing a GET on all my tasks with a certain list id. Here the request : `http://127.0.0.1:8090/api/collections/tasks/records?page=1&perPage=100&filter=list%3Djbgzcemwzgpwtcs` but I get `Invalid filter parameters.`. Do you have any idea why ? Maybe it sounds stupid but I don't how else I can do this

**Reply by ganigeorgiev** [reply]
@Penkie The id need to be a string, .aka quoted with single or double quotes, otherwise internally PocketBase will try to look for a field named "jbgzcemwzgpwtcs" and since it doesn't exist it returns an invalid filter error. Or in other words, your GET url should be:

```
http://127.0.0.1:8090/api/collections/tasks/records?page=1&perPage=100&filter=list%3D%22jbgzcemwzgpwtcs%22
```

**Reply by Penkie** [reply]
Okey understood thank you !
